mjpeg-view/src/main/java/com/github/niqdev/mjpeg/MjpegRecordingHandler.java

```diff
@@ -0,0 +1,145 @@
+package com.github.niqdev.ipcam;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.util.Log;
+import android.widget.Toast;
+import com.github.niqdev.mjpeg.OnFrameCapturedListener;
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class MjpegRecordingHandler implements OnFrameCapturedListener {
+    private static final String TAG = "MjpegRecordingHandler";
+    private Context context;
+    private BufferedOutputStream bos;
+    private boolean isRecording = false;
+    private Bitmap lastBitmap = null;
+
+    public MjpegRecordingHandler(Context context) {
+        this.context = context;
+    }
+
+    /**
+     * start recording the live image
+     */
+    public void startRecording() {
+        try {
+            String mjpegFilePath = createMjpegFile().getAbsolutePath();
+            FileOutputStream fos = new FileOutputStream(mjpegFilePath);
+            bos = new BufferedOutputStream(fos);
+            Toast.makeText(context, "start recording, file path is:" + mjpegFilePath, Toast.LENGTH_LONG).show();
+            isRecording = true;
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+    /**
+     * stop recording the live image
+     */
+    public void stopRecording() {
+        isRecording = false;
+    }
+    public boolean isRecording() {
+        return  isRecording;
+    }
+
+    public void setRecording(boolean recording) {
+        isRecording = recording;
+    }
+
+    public Bitmap getLastBitmap() {
+        return lastBitmap;
+    }
+
+    /**
+     * save the last acquired bitmap into jpg file.
+     */
+    public void saveBitmapToFile() {
+        FileOutputStream fos = null;
+        BufferedOutputStream bos = null;
+        String imagePath = createJpgFile().getAbsolutePath();
+        try {
+            fos = new FileOutputStream(imagePath);
+            bos = new BufferedOutputStream(fos);
+            ByteArrayOutputStream jpegByteArrayOutputStream = new ByteArrayOutputStream();
+            lastBitmap.compress(Bitmap.CompressFormat.JPEG, 75, jpegByteArrayOutputStream);
+            byte[] jpegByteArray = jpegByteArrayOutputStream.toByteArray();
+            bos.write(jpegByteArray);
+            bos.flush();
+            Toast.makeText(context, "saved image:" + imagePath, Toast.LENGTH_LONG).show();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Create jpg file in app external cache directory. the directory path is /sdcard/Android/data/com.github.niqdev.ipcam/files
+     *
+     * @return File
+     */
+    private File createJpgFile() {
```

Comment:
nitpick: this looks a c&p.
What about something like `createFile(prefix: String, extension: String) and refactor the 2 methods below
```
private File createJpgFile() {
  return createFile("photo-", ".jpg")
}
private File createMjpegFile() {
  return createFile("video-", ".mjpeg")
}
```

Reply:
good suggestion